### PR TITLE
[MM-40102] Correctly enable apps plugin for E2E tests 

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
`pluginsEnvironment.Activate` only enabled the plugin in the `plugin.Environment`, but doesn't set `config.PluginStates[pluginID]`.  [mattermost/mattermost-plugin-apps/server/httpin/restapi/restapitestlib.go](https://github.com/mattermost/mattermost-plugin-apps/blob/59f85708320bc10326fa0b6fe4a793eb283d8529/server/httpin/restapi/restapitestlib.go#L106)

If the apps plugin updates the server config, it triggers a `syncPluginsActiveState`. Given that `config.PluginStates[pluginID]` is not set, the plugin gets disabled and shut down. This kills the RPC connection leading to an EOF error.

The actual fix is in 935399358efc15607c6b548eef7f8092175b85d7.

I've also added an E2E test that covers this issue.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40102

### CODE REVIEW

**Review by levb — APPROVED** [review]

**Comment by levb** [comment]
Looks good in theory, but still failing

**Comment by hanzei** [comment]
@levb @catalintomai @mickmister CI is green now. This is ready for review.

**Review by larkox — APPROVED** [review]
LGTM

File: server/httpin/restapi/restapitestlib.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func (th *TestHelper) TearDown() {
 
 func Setup(t testing.TB) *TestHelper {
 	os.Setenv("MM_FEATUREFLAGS_APPSENABLED", "true")
-	defer os.Unsetenv("MM_FEATUREFLAGS_APPSENABLED")
+	t.Cleanup(func() { _ = os.Unsetenv("MM_FEATUREFLAGS_APPSENABLED") })
```
Comment (levb) [inline server/httpin/restapi/restapitestlib.go:47]: 👍

**Pull request merged**